gems/aws-sdk-core/lib/aws-sdk-core/json/builder.rb

```diff
@@ -49,6 +55,7 @@ def format(ref, value)
         when MapShape       then map(ref, value)
         when TimestampShape then timestamp(ref, value)
         when BlobShape      then encode(value)
+        when FloatShape     then Util.serialize_number(value)
```

Comment:
Is this serialization only specific to JSON? If so I think it's appropriate to put serialize in builder and deserialize in parser instead of putting this in the global util namespace.

Reply:
This is generic to all protocols (see our V4/hearth usage)